akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -36,6 +36,40 @@ private[akka] final case class Filter[T](p: T ⇒ Boolean, decider: Supervision.
   override def decide(t: Throwable): Supervision.Directive = decider(t)
 }
 
+/**
+ * INTERNAL API
+ */
+private[akka] final case class TakeWhile[T](p: T ⇒ Boolean, decider: Supervision.Decider) extends PushStage[T, T] {
+  var taking = true
+
+  override def onPush(elem: T, ctx: Context[T]): SyncDirective =
+    if (taking && p(elem)) {
+      ctx.push(elem)
+    } else {
+      taking = false
+      ctx.pull
+    }
```

Comment:
Should have similar semantics to `take(n)`, so it should `ctx.pushAndFinish(elem)` instead of keeping to pull. once condition becomes false.

(Remove taking var as well - it is not needed AFAICS)